rpc/types/src/status.rs

```diff
@@ -0,0 +1,182 @@
+///     CORE_RPC_STATUS_BUSY, CORE_RPC_STATUS_NOT_MINING, CORE_RPC_STATUS_OK,
+///     CORE_RPC_STATUS_PAYMENT_REQUIRED, CORE_RPC_STATUS_UNKNOWN
+/// };
+/// use serde_json::to_string;
+///
+/// let unknown = Status::Unknown;
+///
+/// assert_eq!(to_string(&Status::Ok).unwrap(),              r#""OK""#);
+/// assert_eq!(to_string(&Status::Busy).unwrap(),            r#""BUSY""#);
+/// assert_eq!(to_string(&Status::NotMining).unwrap(),       r#""NOT MINING""#);
+/// assert_eq!(to_string(&Status::PaymentRequired).unwrap(), r#""PAYMENT REQUIRED""#);
+/// assert_eq!(to_string(&unknown).unwrap(),                 r#""UNKNOWN""#);
+///
+/// assert_eq!(Status::Ok.as_ref(),              CORE_RPC_STATUS_OK);
+/// assert_eq!(Status::Busy.as_ref(),            CORE_RPC_STATUS_BUSY);
+/// assert_eq!(Status::NotMining.as_ref(),       CORE_RPC_STATUS_NOT_MINING);
+/// assert_eq!(Status::PaymentRequired.as_ref(), CORE_RPC_STATUS_PAYMENT_REQUIRED);
+/// assert_eq!(unknown.as_ref(),                 CORE_RPC_STATUS_UNKNOWN);
+///
+/// assert_eq!(format!("{}", Status::Ok),              CORE_RPC_STATUS_OK);
+/// assert_eq!(format!("{}", Status::Busy),            CORE_RPC_STATUS_BUSY);
+/// assert_eq!(format!("{}", Status::NotMining),       CORE_RPC_STATUS_NOT_MINING);
+/// assert_eq!(format!("{}", Status::PaymentRequired), CORE_RPC_STATUS_PAYMENT_REQUIRED);
+/// assert_eq!(format!("{}", unknown),                 CORE_RPC_STATUS_UNKNOWN);
+///
+/// assert_eq!(format!("{:?}", Status::Ok),              "Ok");
+/// assert_eq!(format!("{:?}", Status::Busy),            "Busy");
+/// assert_eq!(format!("{:?}", Status::NotMining),       "NotMining");
+/// assert_eq!(format!("{:?}", Status::PaymentRequired), "PaymentRequired");
+/// assert_eq!(format!("{:?}", unknown),                 "Unknown");
+/// ```
+#[derive(
+    Copy, Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize,
+)]
+pub enum Status {
+    // FIXME:
+    // `#[serde(rename = "")]` only takes raw string literals?
+    // We have to re-type the constants here...
+    /// Successful RPC response, everything is OK; [`CORE_RPC_STATUS_OK`].
+    #[serde(rename = "OK")]
+    #[default]
+    Ok,
+
+    /// The daemon is busy, try later; [`CORE_RPC_STATUS_BUSY`].
+    #[serde(rename = "BUSY")]
+    Busy,
+
+    /// The daemon is not mining; [`CORE_RPC_STATUS_NOT_MINING`].
+    #[serde(rename = "NOT MINING")]
+    NotMining,
+
+    /// Payment is required for RPC; [`CORE_RPC_STATUS_PAYMENT_REQUIRED`].
+    #[serde(rename = "PAYMENT REQUIRED")]
+    PaymentRequired,
+
+    /// Some unknown other string; [`CORE_RPC_STATUS_UNKNOWN`].
+    ///
+    /// This exists to act as a catch-all if `monerod` adds
+    /// a string and a Cuprate node hasn't updated yet.
+    ///
+    /// The reason this isn't `Unknown(String)` is because that
+    /// disallows [`Status`] to be [`Copy`], and thus other types
+    /// that contain it.
+    #[serde(other)]
+    #[serde(rename = "UNKNOWN")]
+    Unknown,
+}
```

Comment:
## Status 1/3
I was actually re-thinking doing this `String` optimization stuff, but there is no downside, right?

My main concern was that if `monerod` ever adds a new string in the `status` field, this would fail to deserialize, but, `#[serde(other)]` and...

Reply:
Yeah I think it should be fine, I don't think there are any downsides.